File: src/common/py/features/metadata_feature.py
from .project_feature import ProjectFeature


class MetadataFeature(ProjectFeature):
    FEATURE_ID = "metadata"

    def __init__(self):
        super().__init__(
            feature_id=MetadataFeature.FEATURE_ID,
            flags=ProjectFeature.FeatureFlags.OPTIONAL,
            display_name="Metadata",
        )

File: src/common/web/features/MetadataFeature.ts
import { type ProjectFeatureID } from "../data/entities/EntityTypes";
import { ProjectFeature, ProjectFeatureFlags } from "./ProjectFeature";
import { type ProjectFeaturePanelLoader } from "./ProjectFeaturePanelLoader";

export class MetadataFeature extends ProjectFeature {
    public static readonly FeatureID: ProjectFeatureID = "metadata";

    public constructor(panel?: ProjectFeaturePanelLoader) {
        super(MetadataFeature.FeatureID, {
            flags: ProjectFeatureFlags.Optional,
            displayName: "Metadata",
            optionName: "Project metadata",
            panel: panel
        });
    }
}
